deploy/docs/V1_Migration_doc.md

```diff
@@ -0,0 +1,156 @@
+| sumologic.k8sMetadataFilter.cacheTtl 	| fluentd.metadata.cacheTtl 	|
+| sumologic.k8sMetadataFilter.cacheRefresh 	| fluentd.metadata.cacheRefresh 	|
+| deployment.* 	| fluentd.statefulset.* 	|
+| eventsDeployment.* 	| fluentd.eventsStatefulset.* 	|
+
+
+   - `sumologic.kubernetesMeta` and `sumologic.kubernetesMetaReduce` have been removed. The default log format (`fluentd.logs.output.logFormat`) is `fields`, which removes the relevant metadata from the JSON body of the logs, making these configs no longer necessary.
+   - `sumologic.addStream` and `sumologic.addTime` (default values were `true`) have been removed; the default behavior will remain the same. To preserve the behavior of `addStream = false` or `addTime = false`, you can add the following config to the `values.yaml` file:
+
+```yaml
+fluentd:
+  logs:
+    containers:
+      extraFilterPluginConf:
+        <filter **>
+          @type record_modifier
+          remove_keys stream, time
+        </filter>
+```
+
+Until now, Helm users have not been able to modify their Fluentd configuration outside of the specific parameters that we exposed in the `values.yaml` file. Now, we expose the ability to modify the Fluentd configuration as needed. 
+
+Some use-cases include : 
+ - custom log pipelines, 
+ - adding Fluentd filter plugins (ex: fluentd throttle plugin), or 
+ - adding Fluentd output plugins (ex: forward to both Sumo and S3)
+
+The Fluentd `deployments` have been changed to `statefulsets` to support the use of persistent volumes. This will allow better buffering behavior. They also now include `“fluentd”` in their names. This is not a breaking change for Helm users.
+
+
+### How to upgrade
+#### 1. Upgrade to helm chart version `v0.17.1 `
+
+If the user is not already on `v0.17.1` of the helm chart, upgrade to that version first by running the below command.
+
+```bash
+helm upgrade collection sumologic/sumologic --reuse-values --version=0.17.1
+```
+#### 2: Run upgrade script
+
+For Helm users, the only breaking changes are the renamed config parameters.
+For users who use a `values.yaml` file, we will provide a script that customers can run to convert their existing `values.yaml` files into ones that are compatible with the major release. 
+
+The upgrade script is present in the path : `sumologic-kubernetes-collection/deploy/helm/sumologic/`
+
+Run the upgrade script with the below command.
+```bash
+./upgrade-1.0.0.sh values.yaml
+```
+At this point, users can then run: 
+```bash
+helm upgrade collection sumologic/sumologic --reuse-values --version=1.0.0 -f new_values.yaml
+```
+For users who installed without a `values.yaml` file, if they had set any of the above config parameters they will need to re-set those parameters during their upgrade. For example, if the customer had originally run
+```bash
+helm install sumologic/sumologic --name collection --namespace sumologic --set sumologic.accessId=<SUMO_ACCESS_ID> ... --set sumologic.numThreads=16
```

Comment:
Can we add steps using helm get values? that way customers don't have to remember?

Reply:
e.g helm get values, pipe to file, run upgrade script on file then helm upgrade?

Reply:
The `helm get values` lists all the default properties for the subcharts as well. This might be confusing for the user. Also, we will need to test this path.

Reply:
As we discussed, the better approach here will be

get values used in the current release (needs confirmation in helm2) -> `helm get values -o yaml`
Use this as the base to iterate through and map to changes.
The output should contain only properties from original install list